src/test/java/com/google/googleidentity/oauth2/endpoint/AuthorizationEndpointTest.java

```diff
@@ -0,0 +1,189 @@
+        when(request.getParameter(OAuth2ParameterNames.SCOPE)).thenReturn(null);
+
+        StringWriter stringWriter = new StringWriter();
+        PrintWriter writer = new PrintWriter(stringWriter);
+        when(response.getWriter()).thenReturn(writer);
+
+
+        authorizationEndpoint.doGet(request, response);
+
+        assertThat(httpSession.getClientSession().getRequest()).isPresent();
+
+        assertThat(
+                httpSession.getClientSession()
+                        .getRequest().get().getRequestBody().getScopesList())
+                .containsExactlyElementsIn(CLIENT.getScopesList());
+
+    }
+
+    @Test
+    public void testAuthorizationEndpointGet_CorrectRequest_RedirectToConsentAndCorrectRequest()
+            throws ServletException, IOException {
+
+        HttpServletRequest request = mock(HttpServletRequest.class);
+        HttpServletResponse response = mock(HttpServletResponse.class);
+        FakeHttpSession httpSession = new FakeHttpSession();
+
+        httpSession.setAttribute("user_session", userSession);
+
+        when(request.getSession()).thenReturn(httpSession);
+
+        when(request.getParameter(OAuth2ParameterNames.RESPONSE_TYPE)).thenReturn("code");
+        when(request.getParameter(OAuth2ParameterNames.CLIENT_ID)).thenReturn(CLIENTID);
+        when(request.getParameter(OAuth2ParameterNames.REDIRECT_URI)).thenReturn(REDIRECT_URI);
+        when(request.getParameter(OAuth2ParameterNames.SCOPE)).thenReturn("read");
+        when(request.getParameter(OAuth2ParameterNames.STATE)).thenReturn("111");
+
+        StringWriter stringWriter = new StringWriter();
+        PrintWriter writer = new PrintWriter(stringWriter);
+        when(response.getWriter()).thenReturn(writer);
+
+
+        authorizationEndpoint.doGet(request, response);
+
+        verify(response).sendRedirect("/oauth2/consent");
+
+        assertThat(httpSession.getClientSession().getRequest()).isPresent();
+
+        OAuth2Request oauth2Request = httpSession.getClientSession().getRequest().get();
+
+        OAuth2Request.RequestAuth requestAuth = oauth2Request.getRequestAuth();
+        assertThat(requestAuth.getClientId()).isEqualTo(CLIENTID);
+        assertThat(requestAuth.getUsername()).isEqualTo(USERNAME);
+
+        OAuth2Request.RequestBody requestBody = oauth2Request.getRequestBody();
+        assertThat(requestBody.getGrantType()).isEqualTo("authorization_code");
+        assertThat(requestBody.getIsScoped()).isTrue();
+        assertThat(requestBody.getRefreshable()).isTrue();
+        assertThat(requestBody.getResponseType()).isEqualTo("code");
+        assertThat(requestBody.getScopesList()).containsExactlyElementsIn(CLIENT.getScopesList());
+
+        OAuth2Request.AuthorizationResponse authorizationResponse =
+                oauth2Request.getAuthorizationResponse();
+        assertThat(authorizationResponse.getState()).isEqualTo("111");
+        assertThat(authorizationResponse.getRedirectUri()).isEqualTo(REDIRECT_URI);
```

Comment:
I would strongly recommend using prototruth here. https://truth.dev/protobufs

Then, you can construct the entire expected proto, and a single assertion will give you an error message highlighting any diffs.

```
OAuth2Request.Builder expectedRequest = OAuth2Request.newBuilder().setClientId(...).setUserName(...);
expectedRequest.getRequestAuthBuilder().setGrantType(...)...

assertThat(oauth2Request).isEqualTo(expectedRequest.build());
```